build.gradle

```diff
@@ -36,7 +36,7 @@ project.version = "${SRU}-${scmVersion.version}"
 
 java {
     toolchain {
-        languageVersion.set(JavaLanguageVersion.of(8))
```

Comment:
Could move to Java 11, then release, then migrate to Jakarta, release again, to decouple these two breaking changes.

Reply:
Actually, they are coupled since Jakarta EE 10 classes are compiled with Java 11 so Java 11 is the minimal version

Reply:
Sorry, I misread the comment, yes what you propose would work

Reply:
There is value in splitting the tasks Java 11 and Jakarta I think. I would propose to:

- Upgrade the project to Java 11 (including code style/deprecation fixes)
- Release (major version): First Java 11 release still using javax
- Upgrade to Jakarta (your PR)
- Release (major version): Java 11 release, now using Jakarta

This approach would offer enhanced flexibility to the user base.
There may be users welcoming Java 11 but wanting to avoid Jakarta for the time being.

Even Java 11 by itself feels ancient (active support ends this year), we must keep in mind usage is in financial industry mostly which (from my experience) is not an early adopter.
Upgrading to Java 11 should be a well-publicized step. The same goes for the switch to Jakarta. Both are breaking changes.

> So you prefer 2 PRs instead of one? Or 2 different commits for the same PR could be good enough?
Different PRs would be my preference. I don't do releases on this project, so I would leave that up to Sebastian (@zubri).

cheers

Reply:
I understand the concern of the finance market adoption that is normally slow. However, splitting into two major releases for Java 11 and for Jakarta migration might be too much to handle. This, considering that besides the technology upgrades we have the yearly standards update.

Current standard is SRU2022 and we have a commitment to release the new standards SRU2023 by May. The SRU2023 standards go live at SWIFT by November 2023.

So between May and December we already have to maintain two active branches for current SRU2022 and upcoming SRU2023. If on top we added two extra branches for Java 11 and Jakarta, that becomes, 4 active branches to maintain until November 2023.

All in all, my initial idea would be to target these releases:

- SRU2022-9.3.* -> current standard version that we have to maintain until November 2023.
- SRU2022-10.0.0 -> including both the upgrade to Java 11 and the migration to Jakarta. With alpha-1, alpha-2, etc suffixes until it is stable and tested. For early adopters of the migrated packages, for current SRU until November 2023.
- SRU2023-10.1.0 -> By May. For early testers of the new standards along the migration to Java 11 and Jakarta.

This scheme would give current users enough time to update (until November 2023). And for early adopters, they will be able to do it for both the current and the upcoming standard messages.

Also in the above scheme, major version clearly reflects the Java 11 and Jakarta breaking change. While the minor version, as usual, reflects the yearly standards update.

Finally, all this applies for both projects, the Prowide Core and the Prowide ISO 20022 (the one with the higher impact)

Reply:
Moreover from a pure end-user perspective, proposing a version compiled with Java 11 has minimal benefit compared to what is already possible with this lib that is already compatible with a JRE 11 and even a JRE 17.

Reply:
If the project even loosely follows [semantic versioning](https://semver.org/) rules, a breaking change requires a new major. It is not so much a question of benefit of a change but whether it is breaking.

Reply:
Yes, major version would be changed from 9 to 10 to reflect the breaking change. 
The project does follow semantic versioning, with a custom prefix indicating the SRU.

Reply:
I think it is clear to everybody that a major version is needed for those changes, the question is more do you really want to support 2 new major versions (one for Java 11 + one for Jakarta as you proposed) instead of only one? Frameworks like Spring Boot, Quarkus, Camel, and many others, have one major version to do both (Java upgrade and Jakarta migration), as I did for this PR.